rules/namespace.go

```diff
@@ -143,6 +241,51 @@ func NewNamespaces(version int, namespaces *schema.Namespaces) (Namespaces, erro
 	}, nil
 }
 
+// Clone creates a deep copy of this namespace
+func (nss Namespaces) Clone() (Namespaces, error) {
```

Comment:
nit: add a period at the end.

Seems a bit dirty to do it this way. Would you mind doing it properly in a followup PR that properly does deep cloning without going through object->schema->object conversions?